### ISSUE TITLE
PAGER, less and colour output

### ISSUE THREAD

**Issue by tbm** [ledger/ledger#1920]
15:29 < fjb> Hello, I'm having a problem that ledger displays escape characters e.g "ESC[31m". I'm guessing that this is mostly likely due to badly configured terminal. Unfortunately I am lost on where to start troubleshooting it.

...

21:05 < fjb> @tbm Hi, I was here a couple of days ago asking about escape characters and color in my terminal. I've found the culprit. It was the pager 'less' that didn't correctly parse ASCII escape characters. Adding a 'export LESS=" -R "' to my .bashrc solved that problem.

21:15 < sm[m]> did you mention you were using less ? or does ledger call it automatically ?
21:18 < fjb> Ledger calls it automatically, both when using interactive and non-interactive mode. [...]
21:26 < sm[m]> should it call it with -R ? I think so

**Comment by tbm** [comment]
It seems that's already the case:

```
       if (have_less) {
         on(none, "less");
         setenv("LESS", "-FRSX", 0); // don't overwrite
       }
```

So I think the problem is that $PAGER was set and that was without -R.

Maybe a documentation issue.

**Comment by tbm** [comment]
Hmm, but in my case `$PAGER` is `less` without any arguments and it works fine.

**Comment by tbm** [comment]
Actually one thing I noticed that that ledger doesn't use the pager for me anymore. This has been going on for a while but I never had time to investigate.

**Comment by Amooti73** [comment]
I just want to confirm that I have been having a similar problem with ledger 3.1.3 on ubuntu 20.04.
The problem has existed for the last few weeks.  I have tried many terminals and right now I am using uxterm. This problem is seen in the output of my reports and just yesterday, after failing in all my trial and error attempts I installed hledger which is working well.
![img-2020-05-29-134534](https://user-images.githubusercontent.com/28224877/83252253-9b016800-a1b3-11ea-9bb0-3b3ba7c493ed.png)
I have pasted a screenshot of my output. I have been searching for other reports of this problem for about a week and, although I am not sure I have this exact repo installed, I could not resist posting! Sorry I can't be of more help!

**Comment by Amooti73** [comment]
During the course of this month (May, 2020!) ledger (version 3.1.3 on Ubuntu 20.04) has been producing reports with the ansi escape codes instead of the expected color. I have tried various terminals without any success.  I have also read the above comments about less and tried to make some settings like those recommended [here](https://blog.stigok.com/2017/09/19/system-wide-pager-settings-for-less-on-ubuntu.html).  No success.  I am attaching a screenshot which was made using uxterm.
Just yesterday, I installed hledger and it works well so far.
The screenshot shows a simple balance report and you can see that it is using the pager. My efforts have been pretty much trial and error.  This issue is the first relevant post I have seen on the problem I have been experiencing although I have been searching! Sorry I can't be of more help than saying "me, too"!
![img-2020-05-29-143946](https://user-images.githubusercontent.com/28224877/83256066-6c3ac000-a1ba-11ea-853f-2d389b96df35.png)

**Comment by tbm** [comment]
@Amooti73 is `PAGER` set in your environment? What is the value of``TERM`?

**Comment by Amooti73** [comment]
Using the echo command, pager is blank, although I had also run the command

`update-alternatives --display pager`

The output of this was:

`pager - auto mode
  link best version is /usr/bin/less
  link currently points to /usr/bin/less
  link pager is /usr/bin/pager
  slave pager.1.gz is /usr/share/man/man1/pager.1.gz
/bin/more - priority 50
  slave pager.1.gz: /usr/share/man/man1/more.1.gz
/usr/bin/less - priority 77
  slave pager.1.gz: /usr/share/man/man1/less.1.gz
/usr/bin/w3m - priority 25
  slave pager.1.gz: /usr/share/man/man1/w3m.1.gz`

This, of course, is a bit much for my "hobbyist" level!

When I use echo for TERM, the output is xterm

**Comment by tbm** [comment]
@Amooti73 sorry, I meant the output of:

    echo $PAGER

BTW, can you try:

    export TERM=xterm-256color

to see if that improves things.

**Comment by Amooti73** [comment]
Maybe I wasn't clear but the output from

`echo $PAGER`

was blank--no output!

I just added 

`export TERM=xterm-256color`

to my .bashrc and opened a new terminal, but there was no change!

**Comment by tbm** [comment]
Which terminal do you use?

**Comment by Amooti73** [comment]
I am using uxterm.

If you check the next to last paragraph of my second comment, I also gave some reasons for doing so.   Also I have tried the ledger reports in close to a dozen different terminals as well as inside tmux with no luck.  I have been using ledger for about 4 or 5 years with no problem!

**Comment by tbm** [comment]
Can you try:

    export PAGER="pager -R"

**Comment by Amooti73** [comment]
Success!!!!

Wow! Thanks!

**Comment by Amooti73** [comment]
It also works in tmux! I'm not sure why it shouldn't except for my recent experience with Ctrl-<tab> not calling snippets when I use tmux in uxterm.

**Comment by tbm** [comment]
I think this issue was fixed in 21280a9da3ca6ed70a10f95e1ef0717d5f65ee99 although @Amooti73 said that $PAGER was not set.

**Comment by tbm** [comment]
I'm closing this because I believe it's fixed.
